add ECS support (#85)
logstash-plugins/logstash-filter-csv: docs/index.asciidoc

and stores it as individual fields with optionally-specified field names.
This filter can parse data with any separator, not just commas.

[id="plugins-{type}s-{plugin}-ecs_metadata"]
==== Event Metadata and the Elastic Common Schema (ECS)
When ECS compatibility is disabled, the value is stored in the root level.
When ECS is enabled, the value is stored in the `target` field.

> 

Author: I haven't added the table to compare the columns of ECS disabled and ECS v1 because the field name is depending on the value of `target` which has no default value

Reviewer: I understand the intention but from a plugin user perspective the docs seems confusing.
Maybe we should just include the same message as the one that gets logged to explain?

```
The plugin behaves the same regardless of ECS compatibility, except issuing a warning when `target` isn't set.
It is recommended to set the `target` option to avoid potential schema conflicts.
```
(just a draft - probably need rewording if we decide to use it)

> The plugin behaves the same except giving a warning when ECS is enabled and `target` isn't set.
> It is recommended to set the `target` option to avoid potential schema conflicts.

Reviewer 2: ```suggestion
The plugin behaves the same regardless of ECS compatibility, except giving a warning when ECS is enabled and `target` isn't set.

TIP: Set the `target` option to avoid potential schema conflicts.
```

Reviewer 2: I like the way y'all handled this.  You can use a TIP to put more emphasis on the impact of ECS.  If you want to try this, it will render like this: 

<img width="860" alt="Screen Shot 2021-05-28 at 8 01 55 PM" src="https://user-images.githubusercontent.com/35154725/120051764-f7c67700-bfef-11eb-8e01-f8da6becdf5b.png">
